framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,293 @@
+    """
+    inputSpecification = super(Metric, cls).getInputSpecification()
+    FeaturesInput = InputData.parameterInputFactory("Features", contentType=InputData.StringType)
+    FeaturesInput.addParam("type", InputData.StringType)
+    inputSpecification.addSub(FeaturesInput)
+    TargetsInput = InputData.parameterInputFactory("Targets", contentType=InputData.StringType)
+    TargetsInput.addParam("type", InputData.StringType)
+    inputSpecification.addSub(TargetsInput)
+    MetricInput = InputData.parameterInputFactory("Metric", contentType=InputData.StringType)
+    MetricInput.addParam("class", InputData.StringType, True)
+    MetricInput.addParam("type", InputData.StringType, True)
+    inputSpecification.addSub(MetricInput)
+
+    return inputSpecification
+
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag = 'POSTPROCESSOR Metrics'
+    self.dynamic        = False # is it time-dependent?
+    self.features       = None  # list of feature variables
+    self.targets        = None  # list of target variables
+    self.metricsDict    = {}    # dictionary of metrics that are going to be assembled
+    self.pivotParameter = None
+    # assembler objects to be requested
+    self.addAssemblerObject('Metric', 'n', True)
+
+  def _localWhatDoINeed(self):
+    """
+      This method is a local mirror of the general whatDoINeed method.
+      It is implemented by the samplers that need to request special objects
+      @ In , None, None
+      @ Out, dict, dictionary of objects needed
+    """
+    needDict = {'Metrics':[]}
+    for metricName in self.metricsDict.keys():
+      needDict['Metrics'].append((None, metricName))
+    return needDict
+
+  def _localGenerateAssembler(self,initDict):
+    """Generates the assembler.
+      @ In, initDict, dict, init objects
+      @ Out, None
+    """
+    for metricName in  self.metricsDict.keys():
+      self.metricsDict[metricName] = initDict['Metrics'][metricName]
+
+  def inputToInternal(self, currentInputs):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      @ In, currentInputs, list or DataObject, data object or a list of data objects
+      @ Out, inputDict, dict, current inputs dictionary
+    """
+    if type(currentInputs) == dict and 'features' in currentInputs.keys():
+      return currentInputs
+
+    if type(currentInputs) != list:
+      currentInputs = [currentInputs]
+
+    inputDict = {'features':OrderedDict(), 'targets':OrderedDict()}
+    for currentInput in currentInputs:
+      inputType = None
+      if hasattr(currentInput, 'type'):
+        inputType = currentInput.type
+
+      if isinstance(currentInput, Files.File):
+        self.raiseAWarning(self, "Input type '", inputType, "' is not yet implemented. This input will be skipped")
```

Comment:
I thinks this ```If``` statements should end up in error messages since this pp will not ever accept HDF5 and/or files

Reply:
Changed to an error message.